provider/k8s/helpers.go

```diff
@@ -338,3 +340,54 @@ func calculatePodCpuAndMem(m *metricsv1beta1.PodMetrics) (cpu float64, mem float
 	}
 	return toCpuCore(cpuTotal), toMemMB(memTotal)
 }
+
+func aggregateMetricByPeriod(m structs.Metric, period int64) structs.Metric {
+	sort.Slice(m.Values, func(i, j int) bool {
+		return m.Values[i].Time.After(m.Values[j].Time)
+	})
+
+	vs := structs.MetricValues{}
+	for _, v := range m.Values {
+		if len(vs) > 0 && vs[len(vs)-1].Time.Sub(v.Time).Seconds() <= float64(period) {
```

Comment:
Can we make a variable for this check? I found it difficult to read on a first pass.
It's checking if the metric should be grouped, right?

maybe
```go
withinPeriod := len(vs) > 0 && vs[len(vs)-1].Time.Sub(v.Time).Seconds() <= float64(period)
if withinPeriod {
    ...
}
```